features/dd-sdk-android-trace-otel/src/main/kotlin/com/datadog/android/trace/opentelemetry/OtelTracerProvider.kt

```diff
@@ -135,50 +135,49 @@ class OtelTracerProvider(
          * Builds a [TracerProvider] based on the current state of this Builder.
          */
         fun build(): TracerProvider {
-            return executeSafelyOnAndroid23AndBelow(
-                {
-                    val tracingFeature = sdkCore.getFeature(Feature.TRACING_FEATURE_NAME)
-                        ?.unwrap<Feature>()
-                    val internalCoreWriterProvider = tracingFeature as? InternalCoreWriterProvider
-                    if (tracingFeature == null || internalCoreWriterProvider == null) {
-                        sdkCore.internalLogger.log(
-                            InternalLogger.Level.ERROR,
-                            InternalLogger.Target.USER,
-                            { TRACING_NOT_ENABLED_ERROR_MESSAGE }
-                        )
-                    } else {
-                        sdkCore.updateFeatureContext(Feature.TRACING_FEATURE_NAME) {
-                            it[IS_OPENTELEMETRY_ENABLED_CONFIG_KEY] = true
-                            it[OPENTELEMETRY_API_VERSION_CONFIG_KEY] =
-                                BuildConfig.OPENTELEMETRY_API_VERSION_NAME
-                        }
+            return executeIfJavaFunctionPackageExists(
+                sdkCore.internalLogger,
+                TracerProvider.noop()
+            ) {
+                val tracingFeature = sdkCore.getFeature(Feature.TRACING_FEATURE_NAME)
+                    ?.unwrap<Feature>()
+                val internalCoreWriterProvider = tracingFeature as? InternalCoreWriterProvider
+                if (tracingFeature == null || internalCoreWriterProvider == null) {
```

Comment:
checking whether tracingFeature is null may be redundant. tracingFeature can't be null without internalCoreWriterProvider also being

Reply:
not quite, you can have `tracingFeature` which is of type Feature but not implementing the `InternalCoreWrirerProvider`